packages/checkout/sdk/src/smartCheckout/routing/onRamp/onRampRoute.ts

```diff
@@ -13,14 +13,21 @@ export const onRampRoute = async (
   availableRoutingOptions: AvailableRoutingOptions,
   balanceRequirement: BalanceRequirement,
 ): Promise<OnRampFundingStep | undefined> => {
-  if (balanceRequirement.type !== ItemType.ERC20 && balanceRequirement.type !== ItemType.NATIVE) return undefined;
-  const { required, current, delta } = balanceRequirement;
+  // Only check for ERC20 and NATIVE
+  if (![ItemType.ERC20, ItemType.NATIVE].includes(balanceRequirement.type)) return undefined;
+
+  const { required, current, delta } = balanceRequirement as BalanceNativeRequirement | BalanceERC20Requirement;
 
   let hasAllowList = false;
   const onRampAllowList = await allowListCheckForOnRamp(config, availableRoutingOptions);
 
   onRampAllowList.forEach((token) => {
-    if (token.address === required.token?.address) {
+    if (!token.address) return;
+    if (!required.token) return;
+    if (
+      isNativeToken(required.token.address)
+      || token.address.toLowerCase() === required.token.address?.toLowerCase()
```

Comment:
@imx-mikhala and I had a chat about token address comparisons, this is also a good example of where a util helper to compare token address as case insensitive check would be handy.

Reply:
lets fix that as part of `WT-1957`